x/staking/keeper/validator.go

```diff
@@ -479,7 +491,7 @@ func (k Keeper) SetUnbondingValidatorsQueue(ctx context.Context, endTime time.Ti
 
 // InsertUnbondingValidatorQueue inserts a given unbonding validator address into
 // the unbonding validator queue for a given height and time.
-func (k Keeper) InsertUnbondingValidatorQueue(ctx context.Context, val types.Validator) error {
+func (k *Keeper) InsertUnbondingValidatorQueue(ctx context.Context, val types.Validator) error {
 	addrs, err := k.GetUnbondingValidators(ctx, val.UnbondingTime, val.UnbondingHeight)
```

Comment:
Why are you not using the cache there for `GetUnbondingValidators` ?

Same comment as above, could encapsulate the cache logic directly inside this method to make sure we always hit the cache if it has been initialized